provider/src/iosMain/kotlin/at/asitplus/crypto/provider/os/IOSKeychainProvider.kt

```diff
@@ -0,0 +1,416 @@
+import platform.Security.SecKeyRefVar
+import platform.Security.errSecItemNotFound
+import platform.Security.errSecSuccess
+import platform.Security.kSecAccessControlApplicationPassword
+import platform.Security.kSecAccessControlBiometryAny
+import platform.Security.kSecAccessControlDevicePasscode
+import platform.Security.kSecAccessControlPrivateKeyUsage
+import platform.Security.kSecAccessControlUserPresence
+import platform.Security.kSecAttrAccessControl
+import platform.Security.kSecAttrAccessible
+import platform.Security.kSecAttrAccessibleAfterFirstUnlock
+import platform.Security.kSecAttrAccessibleAfterFirstUnlockThisDeviceOnly
+import platform.Security.kSecAttrAccessibleAlways
+import platform.Security.kSecAttrAccessibleAlwaysThisDeviceOnly
+import platform.Security.kSecAttrAccessibleWhenUnlocked
+import platform.Security.kSecAttrAccessibleWhenUnlockedThisDeviceOnly
+import platform.Security.kSecAttrApplicationTag
+import platform.Security.kSecAttrIsPermanent
+import platform.Security.kSecAttrKeyClass
+import platform.Security.kSecAttrKeyClassPrivate
+import platform.Security.kSecAttrKeyClassPublic
+import platform.Security.kSecAttrKeySizeInBits
+import platform.Security.kSecAttrKeyType
+import platform.Security.kSecAttrKeyTypeEC
+import platform.Security.kSecAttrKeyTypeRSA
+import platform.Security.kSecAttrLabel
+import platform.Security.kSecAttrTokenID
+import platform.Security.kSecAttrTokenIDSecureEnclave
+import platform.Security.kSecClass
+import platform.Security.kSecClassKey
+import platform.Security.kSecKeyOperationTypeSign
+import platform.Security.kSecMatchLimit
+import platform.Security.kSecMatchLimitOne
+import platform.Security.kSecPrivateKeyAttrs
+import platform.Security.kSecPublicKeyAttrs
+import platform.Security.kSecReturnRef
+import platform.Security.kSecUseAuthenticationContext
+import platform.Security.kSecUseAuthenticationUI
+import platform.Security.kSecUseAuthenticationUIAllow
+import platform.Security.kSecUseAuthenticationUISkip
+import platform.Security.kSecUseOperationPrompt
+import secKeyAlgorithm
+
+val keychainThreads = newFixedThreadPoolContext(nThreads = 4, name = "iOS Keychain Operations")
+
+private fun isSecureEnclaveSupportedCurve(c: SigningKeyConfiguration.AlgorithmSpecific): Boolean {
+    if (c !is SigningKeyConfiguration.ECConfiguration) return false
+    return when (c.curve) {
+        ECCurve.SECP_256_R_1, ECCurve.SECP_384_R_1 -> true
+        else -> false
+    }
+}
+
+private object KeychainTags {
+    private val tags by lazy {
+        val bundleId = NSBundle.mainBundle.bundleIdentifier
+            ?: throw UnsupportedCryptoException("Keychain access is unsupported outside of a Bundle")
+        Pair("kmp-crypto-privatekey-$bundleId", "kmp-crypto.publickey-$bundleId")
+    }
+    val PRIVATE_KEYS get() = tags.first
+    val PUBLIC_KEYS get() = tags.second
+}
+
+class iosTPMConfiguration internal constructor() : PlatformSigningKeyConfiguration.TPMConfiguration() {
```

Comment:
here, tpm makes even less sense, because only parts are secure-enclace protected. but SE (in short for Secure Element) is at least somewhat sensible, because as soon as biometrics are involved, an SE in back in the loop